src/main/java/jcifs/smb/SmbFile.java

```diff
@@ -0,0 +1,2380 @@
+            }
+            else {
+                if ( replace ) {
+                    // TRANS2_SET_FILE_INFORMATION does not seem to support the SMB1 RENAME_INFO
+                    throw new SmbUnsupportedOperationException("Replacing rename only supported with SMB2");
+                }
+                sh.send(new SmbComRename(sh.getConfig(), getUncPath(), dest.getUncPath()), new SmbComBlankResponse(sh.getConfig()));
+            }
+
+            this.attrExpiration = this.sizeExpiration = 0;
+        }
+        catch ( CIFSException e ) {
+            throw SmbException.wrap(e);
+        }
+    }
+
+
+    void copyRecursive ( SmbFile dest, byte[][] b, int bsize, WriterThread w, SmbTreeHandleImpl sh, SmbTreeHandleImpl dh ) throws CIFSException {
+        if ( isDirectory() ) {
+            SmbCopyUtil.copyDir(this, dest, b, bsize, w, sh, dh);
+        }
+        else {
+            SmbCopyUtil.copyFile(this, dest, b, bsize, w, sh, dh);
+        }
+
+        dest.clearAttributeCache();
+    }
+
+
+    /**
+     * 
+     */
+    void clearAttributeCache () {
+        this.attrExpiration = 0;
+        this.sizeExpiration = 0;
+    }
+
+
+    @Override
+    public void copyTo ( SmbResource d ) throws SmbException {
+        if ( ! ( d instanceof SmbFile ) ) {
+            throw new SmbException("Invalid target resource");
+        }
+        SmbFile dest = (SmbFile) d;
+        try ( SmbTreeHandleImpl sh = ensureTreeConnected();
+              SmbTreeHandleImpl dh = dest.ensureTreeConnected() ) {
+            if ( !exists() ) {
+                throw new SmbException(NtStatus.NT_STATUS_OBJECT_NAME_NOT_FOUND, null);
+            }
+
+            /*
+             * Should be able to copy an entire share actually
+             */
+            if ( this.fileLocator.getShare() == null || dest.getLocator().getShare() == null ) {
+                throw new SmbException("Invalid operation for workgroups or servers");
+            }
+
+            /*
+             * It is invalid for the source path to be a child of the destination
+             * path or visa versa.
+             */
+            if ( this.fileLocator.overlaps(dest.getLocator()) ) {
+                throw new SmbException("Source and destination paths overlap.");
+            }
+
+            WriterThread w = new WriterThread();
+            w.setDaemon(true);
+
+            try {
+                w.start();
+                // use commonly acceptable buffer size
+                int bsize = Math.min(sh.getReceiveBufferSize() - 70, dh.getSendBufferSize() - 70);
+                byte[][] b = new byte[2][bsize];
+                copyRecursive(dest, b, bsize, w, sh, dh);
+            }
+            finally {
+                w.write(null, -1, null);
+                w.interrupt();
+                try {
+                    w.join();
+                }
+                catch ( InterruptedException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)